pkg/audit/eventformatter/formatters/tenant.go

```diff
@@ -0,0 +1,162 @@
+	}
+
+	return fmt.Sprintf("“%s“ bounded tenant “%s“ to cluster “%s”",
+		event.Metadata["x-auth-email"], tenant.Name, cluster.DisplayName), nil
+}
+
+func (f *tenantEventFormatter) getFormattedDetailsTenantUpdated(ctx context.Context, event *esApi.Event, eventData *eventdata.TenantUpdated) (string, error) {
+	tenantSnapshot, err := f.CreateSnapshot(ctx, projectors.NewTenantProjector(), &esApi.EventFilter{
+		MaxTimestamp: timestamppb.New(event.GetTimestamp().AsTime().Add(time.Duration(-1) * time.Microsecond)), // exclude the update event
+		AggregateId: &wrapperspb.StringValue{Value: event.AggregateId}},
+	)
+	if err != nil {
+		return "", err
+	}
+	oldTenant, ok := tenantSnapshot.(*projections.Tenant)
+	if !ok {
+		return "", esErrors.ErrInvalidProjectionType
+	}
+
+	var details strings.Builder
+	details.WriteString(fmt.Sprintf("“%s“ updated the Tenant", event.Metadata["x-auth-email"]))
+	f.AppendUpdate("Name", eventData.Name.Value, oldTenant.Name, &details)
+	return details.String(), nil
+}
+
+func (f *tenantEventFormatter) getFormattedDetailsTenantDeleted(ctx context.Context, event *esApi.Event) (string, error) {
+	tenantSnapshot, err := f.CreateSnapshot(ctx, projectors.NewTenantProjector(), &esApi.EventFilter{
+		MaxTimestamp: event.GetTimestamp(),
+		AggregateId: &wrapperspb.StringValue{Value: event.AggregateId}},
+	)
+	if err != nil {
+		return "", err
+	}
+	tenant, ok := tenantSnapshot.(*projections.Tenant)
+	if !ok {
+		return "", esErrors.ErrInvalidProjectionType
+	}
+
+	return fmt.Sprintf("“%s“ deleted tenant “%s“", event.Metadata["x-auth-email"], tenant.Name), nil
+}
+
+func (f *tenantEventFormatter) getFormattedDetailsTenantClusterBindingDeleted(ctx context.Context, event *esApi.Event) (string, error) {
+	eventFilter := &esApi.EventFilter{MaxTimestamp: event.GetTimestamp()}
+	eventFilter.AggregateId = &wrapperspb.StringValue{Value: event.AggregateId}
+	tcbSnapshot, err := f.CreateSnapshot(ctx, projectors.NewTenantClusterBindingProjector(), eventFilter)
+	if err != nil {
+		return "", err
+	}
+	tcb, ok := tcbSnapshot.(*projections.TenantClusterBinding)
+	if !ok {
+		return "", esErrors.ErrInvalidProjectionType
+	}
+	eventFilter.AggregateId = &wrapperspb.StringValue{Value: tcb.TenantId}
+	tenantSnapshot, err := f.CreateSnapshot(ctx, projectors.NewTenantProjector(), eventFilter)
+	if err != nil {
+		return "", err
+	}
+	tenant, ok := tenantSnapshot.(*projections.Tenant)
+	if !ok {
+		return "", esErrors.ErrInvalidProjectionType
+	}
+	eventFilter.AggregateId = &wrapperspb.StringValue{Value: tcb.ClusterId}
+	clusterSnapshot, err := f.CreateSnapshot(ctx, projectors.NewClusterProjector(), eventFilter)
+	if err != nil {
+		return "", err
+	}
+	cluster, ok := clusterSnapshot.(*projections.Cluster)
+	if !ok {
+		return "", esErrors.ErrInvalidProjectionType
+	}
+
+	return fmt.Sprintf("“%s“ deleted the bound between cluster “%s“ and tenant “%s“",
+		event.Metadata["x-auth-email"], cluster.DisplayName, tenant.Name), nil
```

Comment:
there is a constant for this, please use that instead of magic strings. replace all occurences please.

Reply:
new stuff learned 🤓. Let me know if I did it correctly.

Reply:
I meant to replace "x-auth-email" with the existing constant in [gateway](https://github.com/finleap-connect/monoskope/blob/c23a26ab384d5c5abc0487589e437281fe051f51/internal/gateway/auth/constants.go#L22). The format strings should stay as they are. Sorry for being not specific enough. :)

Reply:
Done and no worries it was worth the effort :).